library/core/src/cmp.rs

```diff
@@ -289,7 +289,8 @@ pub trait Eq: PartialEq<Self> {
     //
     // This should never be implemented by hand.
     #[doc(hidden)]
-    #[no_coverage] // rust-lang/rust#84605
+    #[cfg_attr(bootstrap, no_coverage)] // rust-lang/rust#84605
```

Comment:
Are these even needed? Since code coverage isn't being run for the bootstrap, we could probably *just* have the `cfg_attr` version and don't need one that explicitly works on the bootstrap.

Reply:
Hmmm, maybe, there are explicit tests that `Eq` doesn't contribute to coverage information, but I'm not sure they're run on the bootstrap build, I'll test and see.

Reply:
Ok, some quick testing reveals that the bootstrap compiler will insert `#[no_coverage]` from its built in `Eq` derive macro but the new core source is the one that adds the `allow_internal_unstable(no_coverage)` to that macro's output (around line 302 of this file).  This seems surprising to me, that the macro is expanded by the compiler's built in expansion logic, but the new source code decides the macro's output's attributes....

Given that we have to keep that attribute at least, I propose to leave them all in for consistency (so the bootstrap core is functionally equivalent to the stage1 core).